### ISSUE TITLE
tab followed by space sends prematurely

### ISSUE THREAD

**Issue by adamdruppe** [CyberShadow/DFeed#101]
It is easy to be typing something an accidentally hit tab, which changes the focus to the "Send" button, then some other text (that does nothing), then a space which sends the post.

I suggest just swapping the tab order of "Send" and "Save and Preview". If you do hit tab then space or enter or whatever, it will do the relatively harmless preview submit instead of a cannot-be-undone partial post submit. If the user did actually intend to save from here, it is easy to quickly proofread and hit tab twice to focus the actual "send" button.

**Comment by CyberShadow** [comment]
Swapping the order of buttons is as easy as swapping two lines. I'm a little concerned though that it's going to mess with the muscle memory of those used to the current order.

**Comment by adamdruppe** [comment]
On Sat, Mar 17, 2018 at 11:28:08PM +0000, Vladimir Panteleev wrote:
> Swapping the order of buttons is as easy as swapping two lines. I'm a little concerned though that it's going to mess with the muscle memory of those used to the current order.

It'd be a minor, temporary annoyance as they get used to it, in exchange for (hopefully) reduced errors overall - including sometimes by me and I usually like tab+enter intentionally to submit.

**Comment by CyberShadow** [comment]
Done!

Any blame on mis-sent posts is on you for the next week or two :)

**Comment by wilzbach** [comment]
FWIW this new change its a bit annoying as now the buttons jump while "Draft saved" is displayed :/

**Comment by CyberShadow** [comment]
Oh, hahah. That's just a bug, I didn't realize I needed to update that JS code. Thanks for the heads up.

**Comment by CyberShadow** [comment]
Fixed. I guess it wasn't as easy as swapping two lines after all :)
